5366 - Fix inconsistent row heights for Datagrid Medium Row Height Setting

vivz753 opened the pull request:
These screenshots are reflective of changes as of 7/14/2021 4:23 PM (EST)

**After changes:**

New
<img width="1266" alt="Screen Shot 2021-07-14 at 1 19 37 PM" src="https://user-images.githubusercontent.com/22609861/125687524-e6ed76f6-4bea-494a-ae2b-a3b8b4e4793a.png">
<img width="1273" alt="Screen Shot 2021-07-14 at 1 19 44 PM" src="https://user-images.githubusercontent.com/22609861/125687527-47b80896-2ebf-4597-947c-eb16b1938c1c.png">

Classic
<img width="1273" alt="Screen Shot 2021-07-14 at 1 19 18 PM" src="https://user-images.githubusercontent.com/22609861/125687513-e73ca875-8997-4a49-8889-802404520cf2.png">
<img width="1273" alt="Screen Shot 2021-07-14 at 1 19 25 PM" src="https://user-images.githubusercontent.com/22609861/125687520-72a87cae-e762-4975-9543-2051bec1d2d0.png">

**Before changes:**

New
<img width="1277" alt="Screen Shot 2021-07-14 at 1 21 03 PM" src="https://user-images.githubusercontent.com/22609861/125687674-a045b2d1-a3e1-49bd-b581-7bed2ae1f85c.png">
<img width="1272" alt="Screen Shot 2021-07-14 at 1 21 09 PM" src="https://user-images.githubusercontent.com/22609861/125687675-94728c2a-6829-4229-af94-07309c7ccdaa.png">

Classic
<img width="1274" alt="Screen Shot 2021-07-14 at 1 20 46 PM" src="https://user-images.githubusercontent.com/22609861/125687666-6d883592-6d47-484f-81b5-1ae965134ebb.png">
<img width="1269" alt="Screen Shot 2021-07-14 at 1 20 54 PM" src="https://user-images.githubusercontent.com/22609861/125687671-7d3709b4-5b36-439b-8260-c3f56b5b51e2.png">



**Explain the _details_ for making this change. What existing problem does the pull request solve?**
<!--
Example: When "Adding a function to do X",
explain why it is necessary to have a way to do X.
-->
- Changed the min-height for the .datagrid.medium-rowheight .datagrid.cell-wrapper from 40px to 46px; this makes the first 2 columns match the other column's row heights

**Related github/jira issue (required)**:
<!--
Provide a link to the related issue(s) to this Pull Request;
auto-closing github issues if necessary (example: "Closes #100")
-->
- Closes #5366 

**Steps necessary to review your pull request (required)**:
<!--
Include:
- commands you ran and their output
- screenshots / videos
- test scenarios
-->
- `npm run start && npm run watch` then go to http://localhost:4000/components/targeted-achievement/example-datagrid.html?theme=new&mode=light&colors=2578a9
- IMPORTANT: This is only applicable for the `New` version
- click the `...` on the top right of the chart, and click `Medium`

**Included in this Pull Request**:
- [ ] An e2e or functional test for the bug or feature.
- [x] A note to the change log.

<!-- After submitting your PR, please check back to make sure tests pass on the github actions. -->

tmcconechy reviewed (changes requested):
@vivz753 I think we need to change this by changing the `chart-completion-target` rules inside the datagrid and not changing the row height (those are locked in by design and could cause other issues. (This is why the regression test is failing)

So rather than change it this way try and adjust the css for `.chart-completion-target` there could be a few rules. Also i see some mismatch on the heights in normal view where the 2 bars are not the same size.

![Screen Shot 2021-07-14 at 9 50 18 AM](https://user-images.githubusercontent.com/814283/125633618-b3932e47-ee46-4d49-99f8-cfed46505b16.png)

vivz753 commented on src/components/datagrid/_datagrid-new.scss:
```diff
@@ -250,8 +250,6 @@
   }
 
   > tbody > tr .datagrid-cell-wrapper {
-    padding: 7px 16px 4px !important;
```
I got rid of this because it doesn't really make sense to have so much padding. This amount of padding makes it so that the right rows on Medium are basically the same size as on the Large setting

vivz753 commented on src/components/datagrid/_datagrid.scss:
```diff
@@ -2413,7 +2413,8 @@ $datagrid-small-row-height: 25px;
     .chart-completion-target {
-      height: 25px;
+      height: 28px; // makes the target bar match the height of progress bar for Large and Medium
+      border-radius: 2px; // round out the corners of the gray target bar
```
Will remove comments after approval -- just in case reviewers are curious why I made these changes

vivz753 commented:
NOTE: fixing the mismatch of bar heights for New Large/Medium caused the Classic Large/Medium to now be mismatched; height for the gray bar needs to be 28px for New, 26px for Classic. 

See here:

`.chart-completion-target` height is set to 28px

New:
<img width="225" alt="Screen Shot 2021-07-14 at 12 20 15 PM" src="https://user-images.githubusercontent.com/22609861/125680422-36e826df-bd20-457e-b864-85c30191c0ce.png">
Classic:
<img width="228" alt="Screen Shot 2021-07-14 at 12 20 26 PM" src="https://user-images.githubusercontent.com/22609861/125680423-bfb51404-436d-4c55-b551-2de1360a3a59.png">

@tmcconechy Should this be tackled by creating overrides in the theme*.scss files? 


**After changes:**

New Medium
<img width="1437" alt="Screen Shot 2021-07-14 at 11 49 56 AM" src="https://user-images.githubusercontent.com/22609861/125677090-990f5665-a57c-4d7f-a311-3d0b135a4a1a.png">

New Large
<img width="1434" alt="Screen Shot 2021-07-14 at 11 54 56 AM" src="https://user-images.githubusercontent.com/22609861/125677096-172fe122-12dd-406c-8e29-de01cf6f03b9.png">

Classic Medium
<img width="1433" alt="Screen Shot 2021-07-14 at 12 17 49 PM" src="https://user-images.githubusercontent.com/22609861/125680084-908730bb-41c2-468a-ade7-5bf56e236993.png">

Classic Large
<img width="1430" alt="Screen Shot 2021-07-14 at 12 16 54 PM" src="https://user-images.githubusercontent.com/22609861/125680031-7e687331-1210-4bd6-91e3-e78af4b12d6c.png">


**Before changes:**

New Medium
<img width="1431" alt="Screen Shot 2021-07-14 at 11 51 38 AM" src="https://user-images.githubusercontent.com/22609861/125676849-24d0ec93-a9d9-4c55-9f80-6600d1d52e26.png">

New Large
<img width="1434" alt="Screen Shot 2021-07-14 at 11 51 22 AM" src="https://user-images.githubusercontent.com/22609861/125676842-91ff678a-8df8-4138-b37e-d24496e25ddc.png">

Classic Medium
<img width="1431" alt="Screen Shot 2021-07-14 at 11 52 40 AM" src="https://user-images.githubusercontent.com/22609861/125676853-22afcaba-82e4-4639-a6c4-5c3cfc327766.png">

Classic Large
<img width="1434" alt="Screen Shot 2021-07-14 at 11 52 50 AM" src="https://user-images.githubusercontent.com/22609861/125676858-044f91e9-4bb9-4065-8ef6-96f2aff6da98.png">

tmcconechy commented:
Yeah thats the right idea i think these overrides go in 

https://github.com/infor-design/enterprise/blob/main/src/components/datagrid/_datagrid.scss#L1368 (more than one place per row height) for classic theme / row heights
And maybe add some in https://github.com/infor-design/enterprise/blob/main/src/components/datagrid/_datagrid-new.scss for new themes if needed

I think these go under datagrid not in config as its the "variation" thats causing the need for an override

vivz753 commented:
Updated CSS to remain the same for Classic (mostly) in `_datagrid.scss`, and added overrides for New in `_datagrid-new.scss`

In `_datagrid.scss`, Classic style needed a 1px change for the gray target progress bar to match the height of the blue achievement progress bar

**After changes:**

New Medium
<img width="219" alt="Screen Shot 2021-07-14 at 1 09 29 PM" src="https://user-images.githubusercontent.com/22609861/125686427-fd1dbc36-6f26-4023-9a2a-ca70c10f668e.png">

New Large
<img width="219" alt="Screen Shot 2021-07-14 at 1 09 18 PM" src="https://user-images.githubusercontent.com/22609861/125686422-4e89ada7-77c9-4ffc-86bf-d1703a3316b0.png">

Classic Medium
<img width="217" alt="Screen Shot 2021-07-14 at 1 09 43 PM" src="https://user-images.githubusercontent.com/22609861/125686431-058fc18a-625a-46ad-b231-27d33a7faa83.png">

Classic Large
<img width="224" alt="Screen Shot 2021-07-14 at 1 09 53 PM" src="https://user-images.githubusercontent.com/22609861/125686433-7a88993f-b504-4844-b1b3-76d1626a3cde.png">

**Before changes:**

New Medium
<img width="217" alt="Screen Shot 2021-07-14 at 1 12 27 PM" src="https://user-images.githubusercontent.com/22609861/125686710-fa2c86e7-3fe7-48a9-870a-bfe16639d15b.png">

New Large
<img width="218" alt="Screen Shot 2021-07-14 at 1 12 36 PM" src="https://user-images.githubusercontent.com/22609861/125686716-e35c5619-24b5-4ba7-bd07-b71172243c90.png">

Classic Medium
<img width="221" alt="Screen Shot 2021-07-14 at 1 13 02 PM" src="https://user-images.githubusercontent.com/22609861/125686717-8471ac8f-9a0c-4962-a459-4c1ae3ce61c7.png">

Classic Large
<img width="221" alt="Screen Shot 2021-07-14 at 1 13 11 PM" src="https://user-images.githubusercontent.com/22609861/125686721-94b7bde9-e582-43aa-96f0-9eb7016585a3.png">

tmcconechy reviewed (approved):
Looks good to me 👍🏻  I updated the baseline image for you since thats a bit tricky right now

vivz753 commented:
Thanks! I'm assuming that's the image for testing purposes? So it doesn't fail the visual regression tests? @tmcconechy

tmcconechy commented:
Yes @vivz753 thats correct. Its all explained here https://github.com/infor-design/enterprise/blob/main/docs/TESTING.md#working-with-visual-regression-tests but it seems a bit much for a first simple issue so i just did it for you quick. But you might get into this in the future.

EdwardCoyle reviewed (approved):
Looks good to me too 👍
